modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/EndoMonoid.kt

```diff
@@ -0,0 +1,13 @@
+package arrow.core.extensions
+
+import arrow.core.Endo
+import arrow.core.compose
+import arrow.core.identity
+import arrow.extension
+import arrow.typeclasses.Monoid
+
+@extension
+interface EndoMonoid<A> : Monoid<Endo<A>> {
+  override fun empty(): Endo<A> = Endo(::identity)
```

Comment:
We should be able to serve a singleton instance of this value to avoid continuous allocations

Reply:
Oh, I see here now how it works. Thanks to everyone for your patience with me.